libs/pika/async_mpi/src/mpi_polling.cpp

```diff
@@ -469,6 +514,48 @@ namespace pika::mpi::experimental {
         }
     }    // namespace detail
 
+    std::uint32_t set_max_requests_in_flight(
+        std::uint32_t N, std::optional<stream_type> s)
+    {
+        if (s == std::nullopt)
```

Comment:
Optional (heh...):
```suggestion
        if (!s)
```
or
```suggestion
        if (!s.has_value())
```
I'm fine with the comparison to `nullopt` as well.